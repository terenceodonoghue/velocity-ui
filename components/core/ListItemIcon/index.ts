export { default } from './ListItemIcon';
export type { ListItemIconProps } from './ListItemIcon.types';
