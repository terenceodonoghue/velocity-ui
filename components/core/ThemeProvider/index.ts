export { default, defaultTheme } from './ThemeProvider';
export type { Theme } from './ThemeProvider.types';
