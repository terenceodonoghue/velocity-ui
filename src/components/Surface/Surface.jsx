import React, { forwardRef } from 'react';
import { arrayOf, node, oneOfType, string } from 'prop-types';
import * as S from './Surface.styles';

export const AppBar = ({ children, ...rest }) => (
  <S.AppBar {...rest}>
    <S.Logo alt="Velocity" src="./logo.svg" />
    <S.Children>{children}</S.Children>
  </S.AppBar>
);

export const Card = forwardRef(({ children, heading, ...rest }, ref) => (
  <S.Card ref={ref} {...rest}>
    {heading && <S.Heading>{heading}</S.Heading>}
    {children}
  </S.Card>
));

AppBar.propTypes = {
  children: oneOfType([arrayOf(node), node]).isRequired,
};

Card.propTypes = {
  children: oneOfType([arrayOf(node), node]).isRequired,
  heading: string,
};

Card.defaultProps = {
  heading: null,
};
