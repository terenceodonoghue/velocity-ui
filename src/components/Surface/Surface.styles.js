import styled from '@emotion/styled';

export const AppBar = styled.div`
  display: flex;
  justify-content: space-between;
  border-bottom: solid 1px #e4e8f0;
  padding: 0 112px;
  height: 80px;
  background-color: ${(props) => props.theme.colors.white};
`;

export const Card = styled.div`
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46, 91, 255, 0.08);
  box-shadow: 0 10px 20px 0 rgba(46, 91, 255, 0.07);
  padding: 32px 24px;
  background-color: ${(props) => props.theme.colors.white};
`;

export const Children = styled.div`
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  margin: 0 -5px;

  & > button {
    margin: 0 5px;
  }

  & > img {
    margin: 0 5px 0 8px;
  }
`;
export const Heading = styled.h3`
  margin: 0 0 26px 0;
  color: ${(props) => props.theme.colors.blueyGrey};
  font-size: 13px;
  font-weight: ${(props) => props.theme.typography.fontWeightRegular};
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;

export const Logo = styled.img`
  cursor: pointer;
`;
