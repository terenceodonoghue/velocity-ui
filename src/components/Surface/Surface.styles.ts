import { css, InterpolationWithTheme } from '@emotion/core';
import { Theme } from '..';

export const appBar: InterpolationWithTheme<Theme> = (theme) => css`
  display: flex;
  justify-content: space-between;
  border-bottom: solid 1px #e4e8f0;
  padding: 0 112px;
  height: 80px;
  background-color: ${theme.colors.white};
`;

export const appBarContent = css`
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  margin: 0 -5px;

  & > button {
    margin: 0 5px;
  }

  & > img {
    margin: 0 5px 0 8px;
  }
`;

export const card: InterpolationWithTheme<Theme> = (theme) => css`
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46, 91, 255, 0.08);
  box-shadow: 0 10px 20px 0 rgba(46, 91, 255, 0.07);
  padding: 32px 24px;
  background-color: ${theme.colors.white};
`;

export const cardHeading: InterpolationWithTheme<Theme> = (theme) => `
  margin: 0 0 26px 0;
  color: ${theme.colors.blueyGrey};
  font-size: 13px;
  font-weight: ${theme.typography.fontWeightRegular};
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;

export const logo = css`
  cursor: pointer;
`;
