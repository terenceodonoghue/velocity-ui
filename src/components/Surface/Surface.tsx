/** @jsx jsx */
import { jsx } from '@emotion/core';
import {
  forwardRef,
  FunctionComponent,
  HTMLAttributes,
  HTMLProps,
} from 'react';
import * as css from './Surface.styles';

interface CardProps extends HTMLProps<HTMLDivElement> {
  heading?: string;
}

export const AppBar: FunctionComponent<HTMLAttributes<HTMLDivElement>> = ({
  children,
  ...props
}) => (
  <div css={css.appBar} {...props}>
    <img css={css.logo} alt="Velocity" src="./logo.svg" />
    <div css={css.appBarContent}>{children}</div>
  </div>
);

export const Card = forwardRef<HTMLDivElement, CardProps>(
  ({ children, heading, ...props }, ref) => (
    <div css={css.card} ref={ref} {...props}>
      {heading && <h3 css={css.cardHeading}>{heading}</h3>}
      {children}
    </div>
  ),
);
