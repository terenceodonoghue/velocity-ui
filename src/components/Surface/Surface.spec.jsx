/** @jsx jsx */
import { jsx } from '@emotion/core';
import * as Surfaces from './Surface';

describe('Surface', () => {
  describe('AppBar', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(withTheme(<Surfaces.AppBar>AppBar Content</Surfaces.AppBar>))
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Card', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(
          withTheme(
            <Surfaces.Card heading="Card Heading">Card Content</Surfaces.Card>,
          ),
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
