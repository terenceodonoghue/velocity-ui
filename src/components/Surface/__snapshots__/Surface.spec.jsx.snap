// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surface AppBar matches snapshot 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: solid 1px #e4e8f0;
  padding: 0 112px;
  height: 80px;
  background-color: #fff;
}

.emotion-0 {
  cursor: pointer;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 -5px;
}

.emotion-2 > button {
  margin: 0 5px;
}

.emotion-2 > img {
  margin: 0 5px 0 8px;
}

<div
  className="emotion-4 emotion-5"
>
  <img
    alt="Velocity"
    className="emotion-0 emotion-1"
    src="./logo.svg"
  />
  <div
    className="emotion-2 emotion-3"
  >
    AppBar Content
  </div>
</div>
`;

exports[`Surface Card matches snapshot 1`] = `
.emotion-2 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: #fff;
}

.emotion-0 {
  margin: 0 0 26px 0;
  color: #8798ad;
  font-size: 13px;
  font-weight: 400;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

<div
  className="emotion-2 emotion-3"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Card Heading
  </h3>
  Card Content
</div>
`;
